contracts/bonding/BondingManager.sol

```diff
@@ -1110,28 +1245,66 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         uint256 currentBondedAmount = del.bondedAmount;
         uint256 currentFees = del.fees;
 
+        uint256 LIP_36_ROUND = roundsManager().LIPUpgradeRounds(36);
+
         // Only will have earnings to claim if you have a delegate
         // If not delegated, skip the earnings claim process
         if (del.delegateAddress != address(0)) {
-            // Cannot claim earnings for more than maxEarningsClaimsRounds
-            // This is a number to cause transactions to fail early if
-            // we know they will require too much gas to loop through all the necessary rounds to claim earnings
-            // The user should instead manually invoke `claimEarnings` to split up the claiming process
-            // across multiple transactions
-            require(_endRound.sub(_lastClaimRound) <= maxEarningsClaimsRounds, "too many rounds to claim through");
+            bool isTranscoder = _delegator == del.delegateAddress;
+
+            Transcoder storage transcoder = transcoders[del.delegateAddress];
 
-            for (uint256 i = startRound; i <= _endRound; i++) {
-                EarningsPool.Data storage earningsPool = transcoders[del.delegateAddress].earningsPoolPerRound[i];
+            // 'pendingFees()' and 'pendingStake()' are O(1) operations after LIP-36
+            if (startRound <= LIP_36_ROUND) {
```

Comment:
Do we need this outer conditional if the inner loop will exit immediately when this is not the case since it would be the case that `startRound > endLoopRound`?

Reply:
yes because we 

1. need a scoped block to avoid stack depth
2. otherwise the require line that checks max claim rounds could underflow.